### PR TITLE
Add crypto:cipher_info/1 and crypto:hash_info/1

### PR DESCRIPTION
**Pull request by essen**
Also adds some more aliases that contain the key length
in their name.

https://bugs.erlang.org/browse/ERL-864

### CODE REVIEW

**Comment by essen** [comment]
Please note there's two open ended questions in the PR (marked with "@todo") and one more question in the ticket.

**Comment by essen** [comment]
I've modified the tests to check that the map returned contains the keys we expect.

File: lib/crypto/c_src/cipher.c [hunk 1]
```diff
@@ -207,6 +219,86 @@ int cmp_cipher_types(const void *keyp, const void *elemp) {
...
+        return enif_make_badarg(env);
+
+    if (FORBIDDEN_IN_FIPS(cipherp))
+        return enif_raise_exception(env, atom_notsup);
+    if ((cipher = cipherp->cipher.p) == NULL)
+        return enif_raise_exception(env, atom_notsup);
+
+    ret = enif_make_new_map(env);
+
+    // @todo This can return NID_undef, should we map this to undefined?
```
Comment (HansN) [inline lib/crypto/c_src/cipher.c]: 'undefined' seems good

File: lib/crypto/c_src/cipher.c [hunk 2]
```diff
@@ -207,6 +219,86 @@ int cmp_cipher_types(const void *keyp, const void *elemp) {
...
+
+        case EVP_CIPH_OFB_MODE:
+            ret_mode = atom_ofb_mode;
+            break;
+
+        case EVP_CIPH_STREAM_CIPHER:
+            ret_mode = atom_stream_cipher;
+            break;
+
+        // @todo Is that a good idea?
```
Comment (HansN) [inline lib/crypto/c_src/cipher.c]: Yes, I think so

Comment (essen) [reply lib/crypto/c_src/cipher.c]: Done both changes and force pushed.

File: lib/crypto/test/crypto_SUITE.erl [hunk 3]
```diff
@@ -665,6 +667,23 @@ rand_plugin_s() ->
...
+    #{type := _,key_length := _,iv_length := _,
+        block_size := _,mode := _} = crypto:cipher_info(aes_128_cbc),
+    {'EXIT',_} = (catch crypto:cipher_info(not_a_cipher)),
+    ok.
+
+%%--------------------------------------------------------------------
+hash_info() ->
+    [{doc, "crypto hash_info testing"}].
+hash_info(Config) when is_list(Config) ->
+    #{type := _,size := _,block_size := _} = crypto:hash_info(sha3_512),
```
Comment (HansN) [inline lib/crypto/test/crypto_SUITE.erl]: sha3_512 is not available in older crypto_libs. Change it to for example sha256 to make the test pass on machines with such libs.

Comment (essen) [reply lib/crypto/test/crypto_SUITE.erl]: Done.

**Comment by HansN** [comment]
Thanks for the PR!

**Pull request merged**